commons/src/main/java/com/powsybl/commons/json/JsonUtil.java

```diff
@@ -582,6 +582,29 @@ public static String getSourceVersion(DatabindContext context, String sourceVers
         return context.getAttribute(sourceVersionAttributeKey) != null ? (String) context.getAttribute(sourceVersionAttributeKey) : null;
     }
 
+    public static <T> T readValueWithContext(DeserializationContext context, JsonParser parser, Class<?> type) {
+        JavaType jType = context.getTypeFactory()
```

Comment:
VERY minor perf optimization: we can move this after the null value check, so that it's not executed for null values.